airbyte/sources/util.py

```diff
@@ -50,6 +51,7 @@ def get_source(
     version: str | None = None,
     pip_url: str | None = None,
     local_executable: Path | str | None = None,
+    docker_executable: bool = False,
```

Comment:
Wdyt of refactoring to take `docker_image` (or `docker_image:tag`) here as a string? 

This would be mutually exclusive with `local_executable` and `pip_url`. If provided, we'd invoke the named image.

Couple things to decide:

1. Should we attempt "docker pull" if the image is not present? (I think 'yes', meaning the user doesn't need to pre-pull the image.)
2. Should we automatically re-pull on every invocation? (I think 'no', meaning the user _does_ need to force a re-pull if they want the latest.)

Curious to your thoughts here. 🤔

Reply:
My thought process around adding a Boolean flag here was to make the end user experience of switching from the default venv execution to Docker seamless. I.e. user can keep all other arguments the same and simply "flip the switch". But looking at it again now I am making the [assumption](https://github.com/airbytehq/PyAirbyte/pull/209/files#diff-c3ea7142407ab026659c30b3c6e11abf4010665385b372f00a994de4c250dccaR137) that an [official Airbyte source](https://hub.docker.com/u/airbyte) will be used. So yes I think refactoring to a ```docker_image``` string will make this more flexible and compatible with any image that follows the Airbyte Protocol!